### PR TITLE
Bootstrap the manager process

### PR DESCRIPTION
**Pull request by marccampbell**

### CODE REVIEW

File: cmd/manager/status.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
```
Comment (emosbaugh) [inline cmd/manager/status.go:6]: ioutil has been deprecated

File: pkg/socket/server.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		c.JSON(http.StatusOK, gin.H{
+			"status": "ok",
+		})
+	})
+
+	return g.RunUnix(socketFile)
+}
+
+func GetSocketPath() (string, error) {
+	tmpDir := os.TempDir()
```
Comment (emosbaugh) [inline pkg/socket/server.go:35]: shouldnt you use /var/run for this?

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: pkg/websocket/server.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		}
+	}
+}
+
+func wsPingHandler(conn *gwebsocket.Conn) func(message string) error {
+	return func(message string) error {
+		logrus.Infof("Received ping message '%s' from server", message)
+		logrus.Infof("Sending pong message '%s'  from server", message)
+		if err := conn.WriteControl(gwebsocket.PongMessage, []byte(message), time.Now().Add(1*time.Second)); err != nil {
+			logrus.Errorf("Failed to send pong message to server: %v", err)
```
Comment (emosbaugh) [inline pkg/websocket/server.go:107]: Why log and not return the error here? Im not saying it is correct. im just assuming the client will not receive the error so im wondering what is the desired behavior here.

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: pkg/websocket/server.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+func pingWSServer(conn *gwebsocket.Conn) error {
+	for {
+		sleepDuration := time.Second * time.Duration(5+rand.Intn(16)) // 5-20 seconds
+		time.Sleep(sleepDuration)
+
+		pingMsg := fmt.Sprintf("%d", rand.Intn(1000))
+		logrus.Infof("Sending ping message '%s' to server", pingMsg)
+
+		if err := conn.WriteControl(gwebsocket.PingMessage, []byte(pingMsg), time.Now().Add(1*time.Second)); err != nil {
+			return errors.Wrap(err, "send ping message")
```
[the remark this answers is not in the document]
Comment (sgalsaleh) [reply pkg/websocket/server.go:88]: this error typically happens when the connection is closed. when this returns, the manager will try to reconnect to kots again.

File: cmd/manager/status.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	"github.com/replicatedhq/embedded-cluster/pkg/socket"
+	"github.com/spf13/cobra"
+)
+
+func StatusCmd(ctx context.Context, name string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: fmt.Sprintf("Get the status of the %s cluster manager", name),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return getServerStatus(ctx)
```
Comment (emosbaugh) [inline cmd/manager/status.go:19]:
```suggestion
			return printServerStatus(ctx)
```

File: cmd/manager/status.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+		},
+	}
+
+	return cmd
+}
+
+func getServerStatus(ctx context.Context) error {
```
Comment (emosbaugh) [inline cmd/manager/status.go:26]:
```suggestion
func printServerStatus(ctx context.Context) error {
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

**Review by emosbaugh — APPROVED** [review]

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**